src/types/c-notification.d.ts

```diff
@@ -0,0 +1,38 @@
+interface IMeta {
```

Comment:
Why did you not maintain the component related interfaces inside the components? I think it would make more sense to keep them in one place.